keeper/accounting.go

```diff
@@ -43,6 +44,27 @@ type AccountingResult struct {
 	NegativeYieldWarning    string
 }
 
+// checkAccountingNotInProgress checks if accounting is currently in progress and returns an error if so.
+// This should be called by operations that modify positions (deposits, withdrawals) to prevent
+// position changes during accounting that could corrupt the accounting cursor state.
+func (k *Keeper) checkAccountingNotInProgress(ctx context.Context) error {
+	cursor, err := k.GetVaultsV2AccountingCursor(ctx)
+	if err != nil {
+		return sdkerrors.Wrap(err, "unable to fetch accounting cursor")
+	}
+	if cursor.InProgress {
+		return sdkerrors.Wrapf(
+			vaultsv2.ErrOperationNotPermitted,
+			"cannot perform this operation while accounting is in progress (started at %s, %d/%d positions processed). "+
+				"Please wait a few minutes for accounting to complete and retry",
+			cursor.StartedAt.String(),
+			cursor.PositionsProcessed,
+			cursor.TotalPositions,
+		)
+	}
+	return nil
+}
+
```

Comment:
I made something like this to block withdrawals, processing withdrawals claims when accounting is in progress. Should probably make sure that all things that blocked functions on accounting use a common function